### ISSUE TITLE
DungeonCrawl 1.16.5-2.3.0 crashes when entering Barathos from Advent of Ascension

### ISSUE THREAD

**Issue by thedarkcolour** [xyroc/DungeonCrawl#85]
Game crashes while generating the terrain for Barathos. Build the ancient teleporter and go in, get this stacktrace.
```
[06Aug2021 17:30:57.012] [Server thread/ERROR] [net.minecraft.server.MinecraftServer/]: Encountered an unexpected exception
net.minecraft.crash.ReportedException: Exception generating new chunk
	at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:137) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:865) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?]
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:118) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?]
	at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242]
Caused by: java.lang.NegativeArraySizeException
	at xiroc.dungeoncrawl.dungeon.DungeonBuilder.generateLayout(DungeonBuilder.java:136) ~[dungeoncrawl:2.3.0]
	at xiroc.dungeoncrawl.dungeon.DungeonBuilder.build(DungeonBuilder.java:105) ~[dungeoncrawl:2.3.0]
	at xiroc.dungeoncrawl.dungeon.Dungeon$Start.generatePieces(Dungeon.java:141) ~[dungeoncrawl:2.3.0]
	at xiroc.dungeoncrawl.dungeon.Dungeon$Start.func_230364_a_(Dungeon.java:130) ~[dungeoncrawl:2.3.0]
	at net.minecraft.world.gen.feature.structure.Structure.func_242785_a(Structure.java:224) ~[?:?]
	at net.minecraft.world.gen.feature.StructureFeature.func_242771_a(SourceFile:35) ~[?:?]
	at net.minecraft.world.gen.ChunkGenerator.func_242705_a(SourceFile:278) ~[?:?]
	at net.minecraft.world.gen.ChunkGenerator.func_242707_a(SourceFile:267) ~[?:?]
	at net.minecraft.world.chunk.ChunkStatus.func_222586_d(ChunkStatus.java:41) ~[?:?]
	at net.minecraft.world.chunk.ChunkStatus.func_223198_a(ChunkStatus.java:198) ~[?:?]
	at net.minecraft.world.server.ChunkManager.lambda$null$18(ChunkManager.java:675) ~[?:?]
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38) ~[datafixerupper-4.0.26.jar:?]
	at net.minecraft.world.server.ChunkManager.lambda$chunkGenerate$20(ChunkManager.java:673) ~[?:?]
	at java.util.concurrent.CompletableFuture.uniCompose(CompletableFuture.java:966) ~[?:1.8.0_242]
	at java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:940) ~[?:1.8.0_242]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:456) ~[?:1.8.0_242]
	at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58) ~[?:?]
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213148_e(SourceFile:94) ~[?:?]
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.func_213145_a(SourceFile:137) ~[?:?]
	at net.minecraft.util.concurrent.DelegatedTaskExecutor.run(SourceFile:105) ~[?:?]
	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402) ~[?:1.8.0_242]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) ~[?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) ~[?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) ~[?:1.8.0_242]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) ~[?:1.8.0_242]
	```

**Comment by xyroc** [comment]
Fixed in version 2.3.2

**Comment by thedarkcolour** [comment]
Thank you!!
